doc: add v2 backing image doc (#1031)
longhorn/website: content/docs/1.8.0/advanced-resources/backing-image/backing-image.md

A QCOW2 or RAW image can be set as the backing/base image of a Longhorn volume, which allows Longhorn to be integrated with a VM like [Harvester](https://github.com/rancher/harvester).

## Create V1 Data Engine Backing Image

### Parameters during creation

#### The data source of a backing image

> You can prepare a v1 data engine backing image using five different kinds of data sources.

Author: fixed, thanks

        backingImageMinNumberOfCopies: "2"
        backingImageNodeSelector: "node1"
        backingImageDiskSelector: "disk1"
        dataEngine: "v1"
      ```

4. Create a PVC with the StorageClass. Then the backing image will be created (with the Longhorn volume) if it does not exist.
5. Longhorn starts to prepare the backing images to disks for the replicas when a volume using the backing image is attached to a node.

#### Notice:
- Please be careful of the escape character `\` when you input a download URL in a StorageClass.

> - Backing image created through StorageClass will follow the same data engine as the volume.

Author: fixed.

##### Use an existing backing Image during volume creation
1. Click **Setting > Backing Image** in the Longhorn UI.
2. Click **Create Backing Image** to create a backing image with a unique name and a valid URL.

> 3. During the volume creation, specify the backing image from the backing image list. The data engine should be the same between the volume and the backing image.

Author: fixed.

#### Notice:
- Users need to make sure the backing image existence when they use UI to create or restore a volume with a backing image specified.
- Before downloading an existing backing image file to the local, users need to guarantee there is a ready file for it.

> - Downloading a v2 data engine backing image is currently not supported.

Author: fixed.

## Create V2 Data Engine Backing Image

> Since v1.8.0, users can create a backing image supported by v2 data engine by specifying `Data Engine` in the YAML, through the UI or StorageClass.

Author: fixed.

### Parameters during creation

> All the parameters are the same as the v1 data engine backing image except `Data Engine`.

Author: fixed.

> #### The data source of a backing image
> You can prepare a v2 data engine backing image using five different kinds of data sources as v1.
> 1. Download a backing image file (using a URL).
> 2. Upload a file from your local machine. This option is available to Longhorn UI users.
> 3. Export an existing in-cluster v1 data engine volume as a backing image.
> 4. Restore a backing image from the backupstore, For more information, see [Backing Image Backup](../backing-image-backup).
> 5. Clone a v1 backing image.

Reviewer: ```suggestion
#### Backing Image Data Sources

You can prepare a V2 Data Engine backing image using any of the supported data sources.

- Download a backing image file (using a URL).
- Upload a file from your local machine. This option is available to Longhorn UI users.
- Export an existing in-cluster V1 Data Engine volume as a backing image.
- Restore a backing image from the backupstore. For more information, see [Backing Image Backup](../backing-image-backup).
- Clone a V1 backing image.
```

Reviewer: We use an unordered list when enumerating options. Ordered lists are usually reserved for procedural steps.

Author: fixed.

> #### Notice:
> - Exporting from a v2 data engine volume is currently not supported
> - Cloning a v2 backing image is currently not supported.
> - Backing up a v2 backing image is currently not supported.
> - For qcow2 image, Longhorn first converts it into raw image and stores the data into v2 data engine backing image due to the v2 data engine limitation.

Reviewer: ```suggestion
#### Notice

- The following operations are currently not supported:
  - Exporting from a V2 Data Engine volume
  - Cloning a V2 backing image
  - Backing up a V2 backing image
- Because of a V2 Data Engine limitation, Longhorn first converts a qcow2 image to a raw image, and then stores the data in a V2 Data Engine backing image.
```

Author: fixed.

#### Notice:
- Users need to make sure the backing image existence when they use UI to create or restore a volume with a backing image specified.
- Before downloading an existing backing image file to the local, users need to guarantee there is a ready file for it.
- Downloading of V2 Data Engine backing images is currently not supported.

## Create a V2 Data Engine Backing Image

Starting v1.8.0, you can create a backing image that is supported by the V2 Data Engine by configuring `Data Engine` in the YAML (through the UI or a StorageClass).

### Parameters During Creation

All parameters are the same as that of the V1 Data Engine backing image, except for `Data Engine`.

#### Backing Image Data Sources

You can prepare a V2 Data Engine backing image using any of the supported data sources.
- Download a backing image file (using a URL).
- Upload a file from your local machine. This option is available to Longhorn UI users.
- Export an existing in-cluster V1 Data Engine volume as a backing image.

> - Restore a backing image from the backupstore. For more information, see [Backing Image Backup](../backing-image-backup).

Reviewer 2: Is restoring the backing image supported in v2? https://github.com/longhorn/longhorn/issues/9992

Author: Yes, it is actually supported naturally
The only part we haven't implemented yet is "**Backing up V2 BackingImage**"
By design, V2 and V1 BackingImage has the same format of the backup in the backupstore, they don't have any different.
Thus, one can restore the BackupBackingImage to V1 or V2 BackingImage in v1.8.0 already.

Reviewer 2: Good. Thanks.

Author: We also tested it in the v2 backing image issue
https://github.com/longhorn/longhorn/issues/6341#issuecomment-2559348262

- Clone a V1 backing image.

#### Notice

- The following operations are currently not supported:
  - Exporting from a V2 Data Engine volume
  - Cloning a V2 backing image
  - Backing up a V2 backing image

> - Because of a V2 Data Engine limitation, Longhorn first converts a qcow2 image to a raw image, and then stores the data in a V2 Data Engine backing image.

Reviewer 2: Can we briefly explain the limitation or reference the corresponding issue for the limitation?

Author: Added
> Unlike the V1 Data Engine, which is file-based, the V2 Data Engine requires Longhorn to store the backing image data in an SPDK logical volume. As a result, for qcow2 images, Longhorn must first convert the qcow2 image to a raw format before storing the data to the V2 Data Engine backing image, enabling it to read the correct data.
